Add support for Transifex localizations

toolness opened the pull request:
Check it out, yo! See the section on localization in the README for more info: https://github.com/toolness/friendlycode/tree/transifex#localization

toolness committed:
Added 'plist' command to bin/build-i18n.js and tests.

This command outputs a plist file for the given i18n bundle module.

Transifex doesn't currently support requirejs i18n bundles, and the
Apple plist format appears to be the closest thing to it.

toolness committed:
Added config.githubBranch to require config, made github URLs more DRY.

toolness committed:
Added a	very early bin/transifex.js and	test suite.

The script doesn't really do anything yet other	than display some
information about the project.

Also added request as a dependency to package.json, as we use it to
communicate with transifex's REST API.

toolness committed:
added more functions to transifex.js with tests.

toolness committed:
wrote network-side part of importing from transifex.

toolness committed:
finished initial implementation of transifex.js.

toolness committed:
added examples/transifex.html and instructions for localization to readme.

toolness committed:
Added --i18n-url option to build-require.js.

toolness committed:
transifex.html now optionally uses transifex-to-requirejs-i18n-server.

This makes it easy for folks to test out their localization in Friendlycode
without needing to clone the git repo.

stenington commented on bin/build-i18n.js:
```diff
@@ -42,19 +56,23 @@ function loadInlineL10nStrings() {
     var defaultValues = InlineL10n.parse(content);
     for (var key in defaultValues) {
       var value = defaultValues[key];
+      var githubUrl = config.githubUrl + '/blob/' + config.githubBranch +
```
I wonder if it's worth mentioning something about the `githubUrl` and `githubBranch` configuration settings in the README? Maybe they're not going to need to be changed often enough to matter though. It seems like you'd only change them if you were localizing a fork with a different set of strings.

toolness replied:
Yeah, this is a bit wonky to me... ideally I think that `build-i18n.js` would introspect into the `.git` directory to discover this information, to avoid the need for these variables in the first place, but there are lots of weird edge cases involved in that.

But wait! Another way to do this might be to use the [repository](https://npmjs.org/doc/json.html#repository) property in `package.json`, and poke at that. It's a far more appropriate location for the information than `require-config.js`!

toolness replied:
Ok, I moved the github-related config information to `package.json` in 8e970e93fac3d7f8d1cada08fda95799020077e5.

stenington replied:
Ah, cool. `package.json` does feel more appropriate.

stenington commented on examples/transifex.html:
```diff
@@ -0,0 +1,33 @@
+    // We need to change the requirejs configuration to redirect
+    // all i18n bundles to transifex.
+    ['fc/nls', 'slowparse-errors/nls'].forEach(function(i18nBundle) {
+      var baseUrl = "http://transifex-to-requirejs-i18n-server.toolness.org/";
```
Maybe it's worth putting a link to the gist with the server code in a comment in here? Again if someone is working on a fork and wants this file to work for their fork, they'd need to change this url to point to a similar service but with their own localizations, right?

stenington commented:
Nice! The README seems accurate to me, and is pretty clear on how to work with Transifex to localize friendlycode. It seems like things get potentially a little confusing if someone wants to fork friendlycode, and then localize that as a separate Transifex project, but I don't think that's a major use case.  

This seems great to me!

toolness committed:
added locale selection widget and querystring arg to examples/transifex.html.

toolness committed:
Moved github-related keys from require-config.js to package.json.

I tried to use the "repository" key in package.json described here:

https://npmjs.org/doc/json.html#repository

The use of 'repository.defaultBranch' is my own invention, though,
which is hopefully acceptable.

stenington commented:
The language dropdown is nice! Setting languages in the browser can be a bit clunky.
